Sources/ArcGISToolkit/Components/Offline/OfflineMapAreasView.swift

```diff
@@ -223,20 +222,16 @@ private class NetworkMonitor: ObservableObject {
     /// The path monitor to observe network changes.
     private let monitor = NWPathMonitor()
     
-    /// A Boolean value indicating whether the device has an internet connection.
-    @Published private(set) var isConnected = true
-    
-    init() {
-        monitor.pathUpdateHandler = { [unowned self] path in
-            DispatchQueue.main.async {
-                self.isConnected = path.status == .satisfied
-            }
+    func startMonitoring(_ onChange: @MainActor @Sendable @escaping (_ isConnected: Bool) -> Void) {
+        monitor.pathUpdateHandler = { path in
+            let isConnected = path.status == .satisfied
+            MainActor.assumeIsolated { onChange(isConnected) }
         }
-        let queue = DispatchQueue(label: "NetworkMonitor")
-        monitor.start(queue: queue)
+        monitor.start(queue: .main)
```

Comment:
Initially I think this change isn't good as we are monitoring some low priority stuff on the main queue.

But I gradually accepted that at some point the Boolean state property needs to be changed on main actor. So just require the escaping closure to run on main actor shouldn't be too much an overhead.